frontend/src/ResourceSearchBar.svelte

```diff
@@ -0,0 +1,109 @@
+<style>
+  select {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  input {
+    background: inherit;
+    color: inherit;
+    border: 1px solid;
+    border-bottom: 1px solid var(--main-fg-color);
+    flex: 1;
+  }
+
+  form {
+    display: inherit;
+    flex: 1;
+  }
+
+  label {
+    display: inherit;
+    flex: 1;
+  }
+
+  .searchbar {
+    display: flex;
+    align-items: left;
+    width: 100%;
+    flex-grow: 1;
+    height: 2em;
+  }
+
+  .optionbar {
+    padding-bottom: 0;
+    padding-top: 0;
+  }
+</style>
+
+<script>
+  import Checkbox from "./Checkbox.svelte";
+  export let rootResource, searchFilter;
+  let searchType, searchQuery, bytesInput, regex, placeholderString, caseIgnore;
+  let searchTypes = ["String", "Bytes"];
+
+  $: if (searchQuery != null && searchType == "Bytes") {
+    try {
+      searchQuery = searchQuery.match(/[0-9a-fA-F]{1,2}/g).join(" ");
```

Comment:
because I'm updating the input reactively, i need to match both 1 and 2 or else the first digit will never make it into the input. This is so we force the bytes format in the input.